src/test/java/uk/gov/companieshouse/api/accounts/service/impl/FilingServiceImplTest.java

```diff
@@ -0,0 +1,461 @@
+
+        when(ixbrlGeneratorMock.generateIXBRL(any(DocumentGeneratorConnection.class)))
+            .thenThrow(new ConnectException());
+
+        assertThrows(ConnectException.class,
+            () -> filingService.generateAccountFiling(transaction, companyAccountEntity));
+    }
+
+    /**
+     * Check environmentReaderMock is called the passed-in number of times and the argument's values
+     * of these calls matches the passed-in args.
+     *
+     * @param numOfTimes The number of times environment reader is be called.
+     * @param args the values the environment reader is called with.
+     */
+    private void verifyEnvironmentReaderNumOfCalls(int numOfTimes, String... args) {
+
+        verify(environmentReaderMock, times(numOfTimes))
+            .getMandatoryString(argCaptor.capture());
+
+        List<String> capturedEnvVariables = argCaptor.getAllValues();
+        Set<String> expectedValues = new HashSet<>(Arrays.asList(args));
+
+        assertTrue(
+            capturedEnvVariables
+                .stream().map(String::toString)
+                .collect(Collectors.toSet())
+                .equals(expectedValues));
+    }
+
+    private void verifyObjectMapperNumOfCalls() throws JsonProcessingException {
+        verify(objectMapperMock, times(1)).writeValueAsString(any(Object.class));
+    }
+
+    private void verifyIxbrlGeneratorNumOfCalls() throws IOException {
+        verify(ixbrlGeneratorMock, times(1)).
+            generateIXBRL(any(DocumentGeneratorConnection.class));
+    }
+
+    /**
+     * Check the returned filing matches the expected filing's data.
+     *
+     * @param filing
+     */
+    private void verifyFilingData(Filing filing) {
+        assertNotNull(filing);
+
+        assertEquals(COMPANY_NUMBER, filing.getCompanyNumber());
+        assertEquals(AccountsType.SMALL_FULL_ACCOUNTS.getAccountType(),
+            filing.getDescriptionIdentifier());
+
+        // TODO: filing.getDescription() is not possible compare since I cannot mock it to get the description
+        // assertEquals(DESCRIPTION, filing.getDescription());
+
+        assertNotNull(filing.getDescriptionValues());
+        assertNotNull(filing.getDescriptionValues().get(PERIOD_END_ON_KEY));
+        assertEquals(AccountsType.SMALL_FULL_ACCOUNTS.getKind(), filing.getKind());
+
+        Data data = filing.getData();
+        assertNotNull(data);
+        assertNotNull(data.getPeriodEndOn());
+
+        List<Link> links = data.getLinks();
+        assertNotNull(links);
+        assertEquals(1, links.size());
+
+        Link link = links.get(0);
+        assertNotNull(link);
+        assertEquals(LINK_RELATIONSHIP, link.getRelationship());
+        assertEquals(IXBRL_LOCATION, link.getHref());
+    }
+
+    /**
+     * Gets json for small full.
+     *
+     * @return
+     */
+    private String getSmallFullJson() {
+        String json =
+            "{\n"
```

Comment:
This will grow massive with all the account types with all the fields. Can the JSON be moved into a file in the test resources folder?

Reply:
2f26d58